src/main/java/org/opensearch/sdk/BaseExtensionRestHandler.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk;
+
+import static org.opensearch.rest.RestStatus.NOT_FOUND;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.opensearch.extensions.rest.ExtensionRestRequest;
+import org.opensearch.extensions.rest.ExtensionRestResponse;
+import org.opensearch.rest.RestHandler.Route;
+
+/**
+ * Provides convenience methods to reduce boilerplate code in an {@link ExtensionRestHandler} implementation.
+ */
+public abstract class BaseExtensionRestHandler implements ExtensionRestHandler {
```

Comment:
Just trying to understand why we need a separate `BaseExtensionRestHandler` class when `ExtensionRestHandler ` can handle the same? Can we move the code there?

Reply:
> Just trying to understand why we need a separate `BaseExtensionRestHandler` class when `ExtensionRestHandler ` can handle the same? Can we move the code there?

Good question.  With JDK8 and `default` methods on interfaces, this could be moved up.  

It's generally my personal style preference to make a pretty sandwich:
 - Clean interface with no more than the class consumer needs (routes and a single `handleRequest()` method.
 - Messy default implementation that no sane user ever wants to deal with.
 - Clean implementation class that's really simple for end users writing the code.

In core, we have a `RestHandler` interface, which is really a functional interface as the only method is `handleRequest()` (and we should probably mimic that part?).

But we also have a `BaseRestHandler` which does the ugly guts of handling the request and only requires the end user to override `prepareRequest()`.

Would really like @dblock opinion on this one.

Reply:
I'm fine with the new abstract class. @dblock if you anything to add

Reply:
I am a Rubyist and thus generally avoid Base* classes because they are a Java thing. I think it's entirely a matter of preference.